definition/rest/v1/rooms.ts

```diff
@@ -38,4 +38,26 @@ export type RoomsEndpoints = {
 			discussion: IRoom;
 		};
 	};
+	'/v1/rooms.export': {
+		POST: (
+			params:
+				| {
+						rid: IRoom['_id'];
+						type: 'file';
+						dateFrom?: string;
+						dateTo?: string;
+						format: 'html' | 'json';
+				  }
+				| {
+						rid: IRoom['_id'];
+						type: 'email';
+						toUsers: string[];
+						toEmails: string[];
+						subject: string;
+						messages: string[];
+				  },
+		) => {
+			missing: string[];
```

Comment:
This return type is only compatible with the 'email' type. When using the 'file' type it'll not have any result.

Reply:
I had to rollback as there is a design limitation on how to represent overloads in REST API operation (i.e. how different payloads and responses are at the same method and path).